### ISSUE TITLE
'xi_ancs' is not defined

### ISSUE THREAD

**Issue by niallomahony93** [ukdsvl/GUML#2]
Hi, I am trying to adapt your code to my data in a python script. 

How are 'xi_ancs' and 'xi_poss' accessed by the 'cost ' function. It seems like they are local to the 'forward' function in the gumlLoss class. I'm finding it difficult to pass this information to the function as it has to work with pymanopt.

**Comment by ukdsvl** [comment]
Please note the following:

1. def cost(R_tensor,L_tensor) requires access to xi_ancs, xi_poss etc, which it obtains via the global variables xi_ancs=embed1.transpose(0,1).detach() , xi_poss=embed2.transpose(0,1).detach() etc. This is required for fixing the remaining network, and learning (R,L) using Riemannian optimization (eg, RCGD) via pymanopt.

2. Like you mentioned, there are also similarly named variables within the gumlLoss class, which is PyTorch based, and is used to compute criterion.

In short, both are required. A clear look at sGUML_CUB_demo.ipynb will clarify this.
